BMM.Core/Implementations/Player/PlaybackHistoryService.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Akavache;
+using BMM.Api.Abstraction;
+using BMM.Api.Implementation.Models;
+using BMM.Core.Helpers;
+using BMM.Core.Implementations.Persistence.Interfaces;
+using BMM.Core.Implementations.Player.Interfaces;
+using BMM.Core.Models.PlaybackHistory;
+using BMM.Core.Models.PlaybackHistory.Interfaces;
+
+namespace BMM.Core.Implementations.Player
+{
+    public class PlaybackHistoryService : IPlaybackHistoryService
+    {
+        public const int MaxEntries = 100;
+        private readonly IBlobCacheWrapper _blobCacheWrapper;
+
+        public PlaybackHistoryService(
+            IBlobCacheWrapper blobCacheWrapper)
+        {
+            _blobCacheWrapper = blobCacheWrapper;
+        }
+
+        public async void AddPlayedTrack(IMediaTrack mediaTrack)
+        {
+            var playbackHistory = await GetPlaybackHistory();
+
+            var lastTrack = playbackHistory
+                .LastOrDefault()
+                ?.MediaTrack;
+
+            if (lastTrack != null && lastTrack.Id == mediaTrack.Id)
+                return;
+
+            if (playbackHistory.Count >= MaxEntries)
+                playbackHistory.RemoveAt(0);
+
+            playbackHistory.Add(new PlaybackHistoryEntry((Track) mediaTrack, DateTime.UtcNow));
+
+            await _blobCacheWrapper.InsertObject(
+                StorageKeys.PlaybackHistory,
+                playbackHistory);
+        }
+
+        public async Task<IEnumerable<IPlaybackHistoryGroup>> GetAll()
+        {
+            var playbackHistory = await GetPlaybackHistory();
+
+            if (!playbackHistory.Any())
+                return Enumerable.Empty<IPlaybackHistoryGroup>();
+
+            var result = playbackHistory
+                .OrderByDescending(x => x.PlayedAtUTC)
+                .GroupBy(l => new
```

Comment:
Hm. We decided to group by day because we have a simple way of showing time information with just introducing a header. But I can easily think of other ways of presenting this information that would require a different grouping.
What I want to say is that the grouping is closely related to the UI and I think it should not bleed into this class. This class should be more independent of how we want to present the information.
I would suggest moving it into the guarded action